src/lazyfree.c

```diff
@@ -177,6 +178,10 @@ void emptyDbAsync(redisDb *db) {
     dict *oldht1 = db->dict, *oldht2 = db->expires;
     db->dict = dictCreate(&dbDictType);
     db->expires = dictCreate(&dbExpiresDictType);
+    if (server.cluster_enabled) {
+         slotToKeyDestroy(db);
+         slotToKeyInit(db);
+    }
```

Comment:
We could just perform the flush like we do in emptyData. We can avoid the free/alloc operations.

```suggestion
    if (server.cluster_enabled) slotToKeyFlush(db);
```

Reply:
slotToKeyFlush won't fix it. We actually call it in emptyData, and emptyData will call emptyDbStructure and emptyDbStructure will call emptyDbAsync, so we already call slotToKeyFlush.

We mainly need this init, to init the clusterDictMetadata, since we create a new dict 
```
db->dict = dictCreate(&dbDictType);

/* Initialize slots-keys map of given db. */
void slotToKeyInit(redisDb *db) {
    db->slots_to_keys = zcalloc(sizeof(clusterSlotToKeyMapping));
    clusterDictMetadata *dictmeta = dictMetadata(db->dict);
    dictmeta->db = db;
}
```

Reply:
Ah, we create a new dict and need to link it.